python/instrumentation/openinference-instrumentation-groq/src/openinference/instrumentation/groq/_request_attributes_extractor.py

```diff
@@ -49,27 +46,115 @@ def get_extra_attributes_from_request(
+        if not hasattr(message, "get"):
+            if isinstance(message, ChatCompletionMessage):
+                message = self._cast_chat_completion_to_mapping(message)
+            else:
+                return
+        if role := message.get("role"):
+            yield (
+                MessageAttributes.MESSAGE_ROLE,
+                role.value if isinstance(role, Enum) else role,
+            )
+
+        if content := message.get("content"):
+            yield (
+                MessageAttributes.MESSAGE_CONTENT,
+                content,
+            )
+
+        if name := message.get("name"):
+            yield MessageAttributes.MESSAGE_NAME, name
+
+        if tool_call_id := message.get("tool_call_id"):
+            yield MessageAttributes.MESSAGE_TOOL_CALL_ID, tool_call_id
+
+        # Deprecated by Groq
+        if (function_call := message.get("function_call")) and hasattr(function_call, "get"):
+            if function_name := function_call.get("name"):
+                yield MessageAttributes.MESSAGE_FUNCTION_CALL_NAME, function_name
+            if function_arguments := function_call.get("arguments"):
+                yield (
+                    MessageAttributes.MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON,
+                    function_arguments,
+                )
+
+        if (tool_calls := message.get("tool_calls"),) and isinstance(tool_calls, Iterable):
+            for index, tool_call in enumerate(tool_calls):
+                if not hasattr(tool_call, "get"):
+                    continue
+                if (tool_call_id := tool_call.get("id")) is not None:
                     yield (
-                        f"{SpanAttributes.LLM_INPUT_MESSAGES}.{index}.{MessageAttributes.MESSAGE_CONTENT}",
-                        content,
+                        f"{MessageAttributes.MESSAGE_TOOL_CALLS}.{index}."
+                        f"{ToolCallAttributes.TOOL_CALL_ID}",
+                        tool_call_id,
                     )
+                if (function := tool_call.get("function")) and hasattr(function, "get"):
+                    if name := function.get("name"):
+                        yield (
+                            f"{MessageAttributes.MESSAGE_TOOL_CALLS}.{index}."
+                            f"{ToolCallAttributes.TOOL_CALL_FUNCTION_NAME}",
+                            name,
+                        )
+                    if arguments := function.get("arguments"):
+                        yield (
+                            f"{MessageAttributes.MESSAGE_TOOL_CALLS}.{index}."
+                            f"{ToolCallAttributes.TOOL_CALL_FUNCTION_ARGUMENTS_JSON}",
+                            arguments,
+                        )
+
+    def _cast_chat_completion_to_mapping(self, message: ChatCompletionMessage) -> Mapping[str, Any]:
+        try:
+            casted_message = dict(message)
+            if (tool_calls := casted_message.get("tool_calls")) and isinstance(
```

Comment:
Instead of casting to `dict`, could we just invoke the attributes directly?
```suggestion
            if (tool_calls := casted_message.tool_calls) and isinstance(
```